config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -112,15 +112,24 @@
                 String configPropName = originalNames.get(beanProp.getName());
                 // Is the property key missing in the config?
                 if (configPropName == null) {
-                    // If so, continue if the field is marked as @{link Optional}
-                    if (isOptionalProperty(clazz, beanProp)) {
-                        continue;
+                    // If the field is not marked as @{link Optional}...
+                    if (!isOptionalProperty(clazz, beanProp)) {
+                        // raise a {@link Missing} exception right here
+                        throw new ConfigException.Missing(beanProp.getName());
+                    }
+                } else {
+                    try {
+                        Object unwrapped = getValue(
```

Comment:
On second thought, would it be acceptable to treat null values as if they were missing, as the spec states? That way, we can avoid exceptions during the normal control flow.